uiowa/uiowa
# Added general_help_text option to the list block and added some default help text to list blocks.

Resolves #5672

This draft PR adds some general help text to the list blocks' configuration panes that I think could be useful for user experience and also will make navigating between editing the blocks and the content within the blocks a lot quicker. 

<img width="451" alt="Screen Shot 2022-09-14 at 11 58 11 AM" src="https://user-images.githubusercontent.com/472923/190216620-8639d5ad-6d71-4664-a846-126422661105.png">

<img width="493" alt="Screen Shot 2022-09-14 at 11 58 27 AM" src="https://user-images.githubusercontent.com/472923/190216681-a98ac603-e0be-4743-a291-8f109db278fe.png">

We've bumped up against folks not having a clear idea about how to manage local people and articles that show up in list blocks. The most common area for confusion is that the user can't directly add a person or post when they edit a block containing these items.

I think this confusion arises from the fact that they're clicking on a post or a person and they expect to be able to edit these items in the same way that clicking a text area  (or any other block like a card, etc) allows them to edit text.

See:

<img width="507" alt="Screen Shot 2022-09-14 at 11 54 58 AM" src="https://user-images.githubusercontent.com/472923/190216196-da9f8545-70a5-42fa-a8f8-a4c6532e2de6.png">

vs

<img width="661" alt="Screen Shot 2022-09-14 at 11 54 48 AM" src="https://user-images.githubusercontent.com/472923/190216226-31732130-e7b1-483d-ae6a-c71a3e59c1a9.png">

While this PR doesn't give them the ability to actually manage people or other content from the block itself, it does provide a front-and-center path to getting to those areas of Drupal and allows them to quickly add a person or article if they're attempting to edit them from the block itself.

Open to feedback or other ideas to improve this.

# How to test

1. Check out this branch
2. On a local site of your choosing, import the configuration: `drush @sandbox.local csim`
3. Add or edit an article list block and note the help text
4. Add or edit a person list block and note the help text
